### PR TITLE
Added support for Unity Catalog `databricks_volume` resource

### PR DESCRIPTION
**Pull request by tanmay-db**
## Changes

Added support for UC Volumes - CRUD APIs. 
 

## Tests

Added unit and integration tests 

- [x] `make test` run locally
- [x] relevant change in `docs/` folder 
- [x] covered with integration tests in `internal/acceptance` 
- [x] relevant acceptance tests are passing
- [x] using Go SDK

### CODE REVIEW

File: catalog/resource_volumes.go
Comment (nfx) [inline catalog/resource_volumes.go:74]: use `catalog.UpdateVolumeRequestContent` as the schema for the request: https://pkg.go.dev/github.com/databricks/databricks-sdk-go@v0.8.1/service/catalog#UpdateVolumeRequestContent

File: catalog/resource_volumes_test.go
Comment (nfx) [inline catalog/resource_volumes_test.go:11]: don't introduce variables for HTTP tests - they're harder to debug as HTTP fixtures.

instead, do inline those variables.

**Review by nfx — CHANGES_REQUESTED** [review]
- it's not using SDK
- add integration tests with steps to update volume name, owner and comments.

File: internal/acceptance/volumes_test.go
Comment (tanmay-db) [inline internal/acceptance/volumes_test.go:8]: We have made a request for this, will check for full lifecycle once we have this capability in our workspace. Skipping for now.

File: catalog/resource_volumes.go
Comment (tanmay-db) [inline catalog/resource_volumes.go:29]: Hi @nkvuong, @nfx suggested to use `catalog.UpdateVolumeRequestContent` but not sure how we can use that here since catalog_name and schema_name for example are required.

File: catalog/resource_volumes.go
Comment (nkvuong) [reply catalog/resource_volumes.go]: I don't think it is possible to move a volume to a new catalog & schema, which is why these parameters are not part of the update request

File: docs/resources/volumes.md
Comment (nkvuong) [reply docs/resources/volumes.md:61]: you could write it as <catalog_name>.<schema_name>.<volume_name>

Comment (tanmay-db) [reply docs/resources/volumes.md]: Thanks updated

File: docs/resources/volumes.md
Comment (tanmay-db) [inline docs/resources/volumes.md:7]: Hi @bogdanrdc, I have used these from the doc, please let me know if these need to be changed or if you have any suggestions. Thank you.

**Review by nkvuong — APPROVED** [review]
went through this with @tanmay-db  and all suggested changes are made. integration tests are also passing

**Comment by tanmay-db** [comment]
Thanks @nkvuong for the review. Hi @nfx, I cannot merge because although we have approval, it says:
`1 review requesting changes and 1 approving review by reviewers with write access. `

Doing a dismiss review to unblock.

File: internal/acceptance/volume_test.go
Comment (tanmay-db) [inline internal/acceptance/volume_test.go:11]: Not sure if there is a better way for filtering specific cloud with UC and UC Volumes enabled.

Comment (pietern) [reply internal/acceptance/volume_test.go:11]: You could set a separate var on the env to filter on. Would this in its current form fail in any of our nightly test environments?

File: catalog/resource_volume.go
Comment (tanmay-db) [reply catalog/resource_volume.go:84]: We need this in update because `name` is updatable and is part of the `full_name` which is used for API calls. If we don't update the FullName then the API calls (post update) will be made to the previous name. Example:

FullName = a.b.c
Name updated from c to d
FullName now is a.b.d but if we don't update it in the Resource Data then the next calls will be made to a.b.c which doesn't exist.

@nkvuong and me discussed this in our meet and one reason why this might not have been required till now is because in other such cases we use `tf:"force_new"` annotation but we shouldn't create and destroy resource when we can just update the ID in resource data. This would need to be done for other resources as well (maybe when they are being migrated).

File: internal/acceptance/volume_test.go
Comment (tanmay-db) [reply internal/acceptance/volume_test.go:11]: > Would this in its current form fail in any of our nightly test environments?
No, this in its current form won't fail any of the nightly test environments because `"TEST_BUCKET"` only exists for one environment currently but this is fragile for short term (in case "TEST_BUCKET" is added to some other environment), in long term I think UC Volumes would be available in UC enabled prod workspaces so we might not required special filtering based on this.

Do you suggest creating a separate variable for this in short term or leave as is? We would need to remove that separate variable once Volumes is default UC enabled so it's an extra overhead.

File: catalog/resource_volume.go
Comment (tanmay-db) [reply catalog/resource_volume.go:84]: Added a comment locally for this, will push with other changes.

Comment (pietern) [reply catalog/resource_volume.go:84]: Understood, thank you. Please add a distilled version of this comment to the call site for future readers.

Comment (tanmay-db) [reply catalog/resource_volume.go:86]: Added the comment.

File: catalog/resource_volume.go
Comment (tanmay-db) [reply catalog/resource_volume.go]: Updated to `VolumeInfo`

**Review by pietern — APPROVED** [review]
Please confirm the integration tests ran against the latest version of this PR.

**Comment by tanmay-db** [comment]
> Please confirm the integration tests ran against the latest version of this PR.

Confirmed - integration tests are passing on latest version of PR.

File: catalog/resource_volume.go
Comment (nfx) [inline catalog/resource_volume.go:16]: please run `make fmt`

File: catalog/resource_volume.go
Comment (nfx) [inline catalog/resource_volume.go:52]: switch the logic a bit for the sake of readability:

```go
if d.Get("owner") == "" {
  return nil
}

var updateVolumeRequestContent catalog.UpdateVolumeRequestContent
...

```

File: catalog/resource_volume.go
Comment (nfx) [reply catalog/resource_volume.go:86]: is there a test that renames it?

File: catalog/resource_volume_test.go
Comment (nfx) [inline catalog/resource_volume_test.go:44]: `				Resource: "/api/2.1/unity-catalog/volumes/" + "testCatalogName.testSchemaName.testName" + "?",`

remove the `" + "`, hinders the greppability

File: docs/resources/volume.md
Comment (nfx) [inline docs/resources/volume.md:38]: include how to create `storage_location` in this example

File: internal/acceptance/volume_test.go
Comment (nfx) [reply internal/acceptance/volume_test.go:11]: we don't `GetEnvOrSkipTest` in the beginning of the test.

File: internal/acceptance/volume_test.go
Comment (nfx) [inline internal/acceptance/volume_test.go:16]: use `{var.STICKY_RANDOM}` instead of `fmt.Sprintf`.

File: internal/acceptance/volume_test.go
Comment (nfx) [reply internal/acceptance/volume_test.go:11]: it's be skipped by `{env.TEST_BUCKET}`

**Review by nfx — COMMENTED** [review]
please address the changes

**Review by nfx — COMMENTED** [review]
please address the changes

File: catalog/resource_volume.go
Comment (tanmay-db) [reply catalog/resource_volume.go:16]: This is not required, it's no-op and fmt test was passed https://github.com/databricks/terraform-provider-databricks/actions/runs/5131170653/jobs/9230902643

**Comment by tanmay-db** [comment]
@nfx we use GetEnvOrSkipTest for other tests: https://github.com/databricks/terraform-provider-databricks/blob/master/internal/acceptance/mws_networks_test.go#L8

what is the reason for:
> we don't GetEnvOrSkipTest in the beginning of the test.

**Comment by tanmay-db** [comment]
> is there a test that renames it?
@nfx yes please take a look at the integration tests that updates it: https://github.com/databricks/terraform-provider-databricks/blob/master/internal/acceptance/volume_test.go#L66

**Pull request merged**
